internal/worker/stateconverter/converter.go

```diff
@@ -71,15 +70,16 @@ func (c *converter) SetUp(ctx context.Context) (watcher.NotifyWatcher, error) {
 }
 
 // Handle implements NotifyWatchHandler's Handle method.  If the change means
-// that the machine is now expected to manage the environment,
-// we throw a fatal error to instigate agent restart.
-func (c *converter) Handle(ctx context.Context) error {
+// that the machine is now expected to manage the environment, we throw a fatal
+// error to instigate agent restart.
+func (c *converter) Handle(_ context.Context) error {
 	c.logger.Tracef("Calling Handle for %s", c.machineTag)
-	results, err := c.machine.Jobs(ctx)
+	isController, err := c.machine.IsController(c.machineTag.Id())
 	if err != nil {
-		return errors.Annotate(err, "can't get jobs for machine")
+		return errors.Trace(err)
 	}
-	if !model.AnyJobNeedsState(results.Jobs...) {
```

Comment:
Only 1 other usage of this remains.

Reply:
Yeah in the internal cloud instance config. That should be done in a separate work, as we need to think about how (and if) the services are going to be used by the internal providers. The `Jobs` (and thereby the `IsController`) used in there is a static information baked into the instance config, however, we're making it sort of a dynamic information by requiring asking the machine service.